src/com/sun/ts/tests/signaturetest/SigTest.java

```diff
@@ -208,13 +207,45 @@ public void signatureTest() throws Fault {
     String packageFile = getPackageFile();
     String testClasspath = testInfo.getTestClasspath();
     String optionalPkgToIgnore = testInfo.getOptionalTechPackagesToIgnore();
+    String jimageDir = testInfo.getJImageDir();
 
     // unlisted optional technology packages are packages for optional
     // technologies that were not specified by the user. We want to
     // ensure there are no full or partial implementations of an
     // optional technology which were not declared.
     ArrayList<String> unlistedTechnologyPkgs = getUnlistedOptionalPackages();
 
+    // If testing with Java 11, extract the JDK's modules so they can be used
+    // on the testcase's classpath.
+    Properties sysProps = System.getProperties();
+    String version = (String) sysProps.get("java.version");
+    if (version.startsWith("11")) {
```

Comment:
Would it be better to just do this for Java 9 and above to future proof this and not require a future change for later JDKs.  

if (!version.startsWith("1.")) {

Reply:
Until we solve https://github.com/eclipse-ee4j/jakartaee-tck/issues/156 (stop checking JDK signatures) we are dependent on specific Java SE versions.  Although whether all future JDKs will work with this code change, I'm not sure.  

My current thinking is that we will need to switch to checking if the Java version is equal/greater than the current base Java SE version (after we solve jakartaee-tck/issues/156).  If that makes sense, we will be saying that the TCK can work with any later JDK version as long as the later JDK version doesn't break compatibility with what the TCK depends on.  In summary, I think that is the best possible and might mean that a particular TCK version only really works with the base Java SE version + the next N JDK versions until some (later JDK) breaking change causes breakage.  So, that later JDK version will not work with the earlier TCK until a new TCK (for a new Jakarta EE release) is released that works with the new JDK version.  This is just my current thinking and the community, you, others need to buy into it when we actually look to resolve jakartaee-tck/issues/156.  Sorry for the interruption and tangent on this pr. :-)

Reply:
I've gone ahead and made Jared's proposed changes.  Even if we move forward with changes related to 156, making this change would remove the need for us to update the driver code for this particular scenario again.